internal/utils/map_get.go

```diff
@@ -0,0 +1,66 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import "fmt"
+
+// Not sure if this is idiomatic go, but I'm trying to make it easier
+// to work with untyped maps such as might be created by unmarshalling
+// some arbitrary JSON
+
+// Access a string value from a map with string keys
+func MapGetStr(m map[string]any, key string) (string, error) {
+	return mapGet[string](m, key)
+}
+
+// Access a bool value from a map with string keys
+func MapGetBool(m map[string]any, key string) (bool, error) {
+	return mapGet[bool](m, key)
+}
+
+// Access an int value from a map with string keys
+func MapGetInt(m map[string]any, key string) (int, error) {
+	return mapGet[int](m, key)
+}
+
+// Access a list in map with string keys
+func MapGetList(m map[string]any, key string) ([]any, error) {
+	return mapGet[[]any](m, key)
+}
+
+// Access a map inside a map with string keys
+func MapGetMap(m map[string]any, key string) (map[string]any, error) {
+	return mapGet[map[string]any](m, key)
+}
+
+func mapGet[T string | bool | int | []any | map[string]any](m map[string]any, key string) (T, error) {
```

Comment:
I'd probably have this as a [public interface](https://go.dev/play/p/UciROptYns7), and move this to `utils/maps`, so we could write something like `maps.Get(...)`

Reply:
I think I tried that but was discourage when I learned that you can't do a generic in a interface method. But still the public methods could be done like this. I'll give it a try.